chromium_src/chrome/browser/ui/media_router/media_router_ui_service_factory.cc

```diff
@@ -0,0 +1,33 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/ui/media_router/media_router_ui_service_factory.h"
+
+#include "brave/common/pref_names.h"
+#include "chrome/browser/media/router/chrome_media_router_factory.h"
+#include "chrome/browser/ui/toolbar/toolbar_actions_model_factory.h"
+#include "chrome/common/pref_names.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+
+#define BuildServiceInstanceFor BuildServiceInstanceFor_ChromiumImpl
+#include "../../../../../../chrome/browser/ui/media_router/media_router_ui_service_factory.cc"  // NOLINT
+#undef BuildServiceInstanceFor
+
+namespace media_router {
+
+KeyedService* MediaRouterUIServiceFactory::BuildServiceInstanceFor(
+    BrowserContext* context) const {
+#if !defined(OS_ANDROID)
```

Comment:
this also needs a comment because I don't understand why we're doing this or why it's happening here because this definitely doesn't seem like the right place. If `kEnableMediaRouter` needs to be kept in sync with `kBraveMediaRouter`, there should be a pref observer for it

Reply:
Comment added: https://github.com/brave/brave-core/pull/10116/files#diff-c1bca8803e527698720c7fb05fa895bbcc5537f9accbf845c48b2709cdecd26eR25